sdk/storage/azblob/container/client.go

```diff
@@ -124,7 +133,7 @@ func (c *Client) URL() string {
 func (c *Client) NewBlobClient(blobName string) *blob.Client {
 	blobName = url.PathEscape(blobName)
 	blobURL := runtime.JoinPaths(c.URL(), blobName)
-	return (*blob.Client)(base.NewBlobClient(blobURL, c.generated().Pipeline(), c.credential()))
+	return (*blob.Client)(base.NewBlobClient(blobURL, c.generated().InternalClient(), c.credential()))
```

Comment:
This will need some refactoring.  The created `*azcore.Client` contains instance-specific content (e.g. `"container.Client"`) which isn't appropriate for other clients.

Reply:
Per discussion, will wait for the fix in https://github.com/Azure/azure-sdk-for-go/issues/20973